### PR TITLE
Retry node pool deletion in autoscaling tests.

### PR DESCRIPTION
**Pull request by bskiba**
**What this PR does / why we need it**:
Currently, when there is already an operation running on the cluster, the node pool deletion will fail. This will in turn make other tests that try to  add extra node pool fail.
With this PR we will retry the node pool deletion operation 3 times over the course of ~15 minutes which will hopefully be enough to finish any operations running on the cluster.
This will deflake autoscaling e2e tests on gke regional clusters. 

**Release note**:
```
NONE
```

@aleksandra-malinowska

### CODE REVIEW

**Comment by aleksandra-malinowska** [comment]
> Currently, when there is already an operation running on the cluster, the node pool deletion will fail. 
Out of curiosity - do we know what are these other operations? Are they supposed to be going on?

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -1304,11 +1304,17 @@ func deleteNodePool(name string) {
...
-		glog.Infof("Error: %v", err)
-	}
-	glog.Infof("Node-pool deletion output: %s", output)
+	wait.ExponentialBackoff(
+		wait.Backoff{Duration: 1 * time.Minute, Factor: float64(3), Steps: 3},
+		func() (bool, error) {
+			output, err := execCmd(getGcloudCommand(args)...).CombinedOutput()
+			if err != nil {
+				glog.Warningf("Error deleting nodegroup - error:%v, output: %s", err, output)
+				return false, nil
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: nit: we should probably fail the test which triggered clean-up after the last attempt failed. This won't prevent disrupting subsequent tests, but if the same one is causing this problem across runs, at least we'll see the trend on the test-grid

**Comment by aleksandra-malinowska** [comment]
One nit and a question to consider, otherwise /lgtm

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 1 above)
Comment (bskiba) [reply test/e2e/autoscaling/cluster_size_autoscaling.go]: Makes sense, will do. Although currently in regional tests I'm seeing time-outs when this happens, as all the subsequent tests wait for a lower number of nodes to be available both at the beginning of the test (5 minutes) and at the end of the test (20 minutes). That's something to consider fixing too (at least skip the second wait if the first one fails), but next PR.

**Comment by aleksandra-malinowska** [comment]
> We have at least master upgrade to expect, right?

Do you mean master upgrade triggered by increasing cluster size (to a larger machine type)?

**Comment by bskiba** [comment]
> Do you mean master upgrade triggered by increasing cluster size (to a larger machine type)?

That's the first example that comes to mind.

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 1 above)
Comment (bskiba) [reply test/e2e/autoscaling/cluster_size_autoscaling.go]: Done.

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 2]
```diff
@@ -509,7 +509,7 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		framework.ExpectNoError(framework.DeleteRCAndPods(f.ClientSet, f.InternalClientset, f.ScalesGetter, f.Namespace.Name, "node-selector"))
 	})
 
-	It("should scale up correct target pool [Feature:ClusterSizeAutoscalingScaleUp]", func() {
+	It("should scale up correct target pool [Feature:ClusterSizeAutoscalingScaleUp1]", func() {
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: is this changed on purpose?

Comment (bskiba) [reply test/e2e/autoscaling/cluster_size_autoscaling.go]: Nah, debug. Removed.

**Pull request merged**